src/Agent.PluginCore/CommandPlugin.cs

```diff
@@ -0,0 +1,317 @@
+            }
+
+            private bool IsMatchInBypassList(Uri input)
+            {
+                string matchUriString = input.IsDefaultPort ?
+                    input.Scheme + "://" + input.Host :
+                    input.Scheme + "://" + input.Host + ":" + input.Port.ToString();
+
+                foreach (Regex r in _regExBypassList)
+                {
+                    if (r.IsMatch(matchUriString))
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        private class CommandPluginClientCertificateManager : IVssClientCertificateManager
+        {
+            private readonly X509Certificate2Collection _clientCertificates = new X509Certificate2Collection();
+            public X509Certificate2Collection ClientCertificates => _clientCertificates;
+            public CommandPluginClientCertificateManager(string clientCertificateArchiveFile, string clientCertificatePassword)
+            {
+                if (!string.IsNullOrEmpty(clientCertificateArchiveFile))
+                {
+                    _clientCertificates.Add(new X509Certificate2(clientCertificateArchiveFile, clientCertificatePassword));
+                }
+            }
+        }
+
+        private AgentCertificateSettings GetCertConfiguration()
+        {
+            bool skipCertValidation = PluginUtil.ConvertToBoolean(this.Variables.GetValueOrDefault("Agent.SkipCertValidation")?.Value);
+            string caFile = this.Variables.GetValueOrDefault("Agent.CAInfo")?.Value;
+            string clientCertFile = this.Variables.GetValueOrDefault("Agent.ClientCert")?.Value;
+
+            if (!string.IsNullOrEmpty(caFile) || !string.IsNullOrEmpty(clientCertFile) || skipCertValidation)
+            {
+                var certConfig = new AgentCertificateSettings();
+                certConfig.SkipServerCertificateValidation = skipCertValidation;
+                certConfig.CACertificateFile = caFile;
+
+                if (!string.IsNullOrEmpty(clientCertFile))
+                {
+                    certConfig.ClientCertificateFile = clientCertFile;
+                    string clientCertKey = this.Variables.GetValueOrDefault("Agent.ClientCertKey")?.Value;
+                    string clientCertArchive = this.Variables.GetValueOrDefault("Agent.ClientCertArchive")?.Value;
+                    string clientCertPassword = this.Variables.GetValueOrDefault("Agent.ClientCertPassword")?.Value;
+
+                    certConfig.ClientCertificatePrivateKeyFile = clientCertKey;
+                    certConfig.ClientCertificateArchiveFile = clientCertArchive;
+                    certConfig.ClientCertificatePassword = clientCertPassword;
+                }
+
+                return certConfig;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        private AgentWebProxySettings GetProxyConfiguration()
```

Comment:
this is consistent with how the libs read the settings, correct?

Reply:
yes, same as node/powershell lib, read setting from variables set by the agent..